### ISSUE TITLE
Framework:  Prepare PR Jobs for CMake minimum version update

### ISSUE THREAD

**Issue by william76** [trilinos/Trilinos#8520]
@trilinos/framework

Description
-----------
As part of #8401, we need to update all the PR test scripts to load CMake >= 3.17.0 in preparation for the minimum-required version switchover that's coming on 2021-01-30.

- This issue is part of #8401

Tasks
-----
We should update the version of CMake that is used by our PR and Test replication jobs to set to the minimum version of CMake over 17.0
See the list in #8401 for the master list.

| Platform(s)           | Available Module                          |
|-----------------------|-------------------------------------------|
| RHEL7 COE + SEMS      | sems-cmake/3.17.1                         |
| Ride/White            | devpacks load 3.12, where is a newer one?<BR> `/home/atdm-devops-admin/tools/ride/cmake-3.17.2/bin/cmake` |
| RHEL7 COE Cloud Nodes | sems-cmake/3.17.1                         |
| Vortex                | cmake/3.18.0                              |

### Update Jenkins Pull Request Jobs
This should be in the `cmake/std/pr_config/pullrequest.ini`
- [x] `Trilinos_pullrequest_clang_10.0.0`
- [x] `Trilinos_pullrequest_clang_7.0.1`
- [x] `Trilinos_pullrequest_clang_9.0.0`
- [x] `Trilinos_pullrequest_cuda_9.2`  _(cmake loaded via devpack)_
- [x] `Trilinos_pullrequest_cuda_10.1.105` _(cmake loaded via devpacks)_
-  ~~`Trilinos_pullrequest_cuda_10.1.243`~~ _(not in pullrequest.ini)_
- [x] `Trilinos_pullrequest_gcc_4.8.4`
- [x] `Trilinos_pullrequest_gcc_4.9.3_SERIAL`
- [x] `Trilinos_pullrequest_gcc_7.2.0`
- [x] `Trilinos_pullrequest_gcc_7.2.0_debug`
- [x] `Trilinos_pullrequest_gcc_7.2.0_serial`
- [x] `Trilinos_pullrequest_gcc_8.3.0`
- [x] `Trilinos_pullrequest_intel_17.0.1`
- [x] `Trilinos_pullrequest_intel_19.0.5`
- [x] `Trilinos_pullrequest_python_3`

### Update PR Replication Scripts
These are in `cmake/std/sems/`
- [x] PullRequestClang10.0.0TestingEnv.sh
- [x] PullRequestClang7.0.1TestingEnv.sh
- [x] PullRequestClang9.0.0TestingEnv.sh
- [x] PullRequestCuda10.1.105TestingEnv.sh
- [x] PullRequestCuda9.2TestingEnv.sh
- [x] PullRequestGCC4.8.4TestingEnv.sh
- [x] PullRequestGCC4.9.3TestingEnv.sh
- [x] PullRequestGCC4.9.3TestingEnvSERIAL.sh
- [x] PullRequestGCC7.2.0SerialTestingEnv.sh
- [x] PullRequestGCC7.2.0TestingEnv.sh
- [x] PullRequestGCC8.3.0TestingEnv.sh
- [x] PullRequestIntel17.0.1TestingEnv.sh
- [x] PullRequestIntel19.0.5TestingEnv.sh
- [x] PullRequestPython3TestingEnv.sh

**2021-01-13 Standup Action Items**
- [x] Based on information provided in [8401][4], there is a CMake 3.17.2 on Ride but it's in the atdm-devops homedir. Since we currently use ninja and cmake out of Ross's home dir the best solution might be to just change the path for CMake to that rather than messing with modules. Update the branch and let the WIP PR #8521 run a cycle to test it out. We won't merge until Jan 31st if it works.
- [x] File a ticket with SEMS (should be routed to Joe) regarding CMake availability on  Ride if the ATDM approach works. Reference the discussion on [TRIL-208][3] that explained the module issues that we had on Ride and why we didn't use the SEMS modules. Also, note that /projects/sems is not mounted on Ride but it may not be worth the time investment to get SEMS modules mounted there at this time. Might also reference `SEHELPD-2284`

### Dependencies
- [x] [Kokkos-Kernels CMake minimum update to 17.0.1][1] ([Kokkos: 3679][2])

### Due Date
- This is due before January 30th, 2021.

FYI @jwillenbring this is from our conversation this morning to track progress towards getting ready to upgrade the minimum CMake requirement.

[1]: https://github.com/kokkos/kokkos-kernels/pull/866
[2]: https://github.com/kokkos/kokkos/pull/3679
[3]: https://sems-atlassian-son.sandia.gov/jira/browse/TRIL-208
[4]: https://github.com/trilinos/Trilinos/issues/8401#issuecomment-736852480

**Comment by william76** [comment]
I've started working on this in a Trilinos branch [fmwk-prepare-cmake-minimum-update][1]. I put the branch on Trilinos so that other people in @trilinos/framework can work on it if needed.

[1]: https://github.com/trilinos/Trilinos/tree/fmwk-prepare-cmake-minimum-update

**Comment by william76** [comment]
@jwillenbring I'm not sure what we should be doing on the CUDA builds since they load devpacks that contain specific versions of CMake. I guess the right thing will be to do something like:

```
module load <devpack>
module unload cmake/a.b.c
module load cmake/x.y.z
```

@ZUUL42 Do you know what the right modules we'd want to use to load the newer CMake on the CUDA jobs?

**Comment by william76** [comment]
TODO:
- [ ] File ticket with SEMS re: cmake availability on Ride / White.  `/projects/sems` does not exist on Ride. Reference JIRA ticket `???`  for SEMS. Should check with Joe.

**Comment by william76** [comment]
@jwillenbring @ZUUL42 @prwolfe @e10harvey 
I ran a sequence of tests to test out the [Replicating Errors][1] script with the CMake 3.17.x updated module use.

I only ran them through the configure step, but they all completed without errors. I didn't run build with that since it'll probably take 24+ hours to complete for the batch. 

I marked off a bunch in #8521 since we don't run those PR's anymore as far as I can tell. We should probably revisit the existence of those scripts at some point to clean house and remove the ones that aren't (__or won't ever be__) used anymore.

PR #8660 is up now too with update to the top-level CMakeLists.txt file to increase the minimum version to 3.17.0.

[1]: https://github.com/trilinos/Trilinos/wiki/Reproducing-PR-Testing-Errors
